### ISSUE TITLE
Complex startup procedure off: Nasal error message

### ISSUE THREAD

**Issue by dany93** [Juanvvc/c172p-detailed#757]
When I launch the aircraft with the "**Start with engine running**" checked (only this start option checked) I have this error message:
`Nasal runtime error: No such member: autostart
  at /home/dany/FG/github/c172p/Nasal/engine.nas, line 185
  called from: /home/dany/fgfs/install/flightgear/fgdata/Nasal/globals.nas, line 119`
I've had it for a long time.
Still, everything works well. It starts with the engine running.

I do not have this error if "**Complex Startup Procedure**" is checked, or both ("**Complex Startup Procedure**" and "**Start with engine running**") are checked.

_If I'm the only one, no big worry, but just in case..._

FG 2016.2.0 git (download and compile v 2.33). Updated 14 May 2016.
Linux Mint 17.

**Comment by gilbertohasnofb** [comment]
It seems related to the if block in `engine.nas`, lines 184-185: 

```
    if (!getprop("/fdm/jsbsim/complex"))
        c172p.autostart(0);
```

In the file `c172p.nas`, line 304, if the option start with engine running is activated (and therefore the property `/fdm/jsbsim/running` is set to true), the autostart function will also be called as `c172p.autostart(0);`. So maybe calling it twice in a row is what is causing this little error? I will try having a look at it.

**Comment by gilbertohasnofb** [comment]
Actually, the error has nothing to do with starting with the engine running. Basically if "complex startup" is deactivated, the error is present _regardless_ of the state of "start with engine running"

**Comment by gilbertohasnofb** [comment]
Hmm, for some reason FG is not happy about that function call of `c172p.autostart(0);` on line 185 of `engine.nas`, but I don't understand why... surely one can call that function from another `.nas` file, right?

**Comment by dany93** [comment]
And what is the use of "Complex Startup Procedure" option choice ? I feel that It could be always ON, that allows every possibilities: complex (of course), "Start with engine running" (if checked) and "autostart" (if "Start with engine running" is unchecked). 
My question is: could "Complex Startup Procedure" be always ON (not an option)? Deleted from the list. One would have as choices "Start with engine running" checked or not.
Or maybe I've not seen something?

**Comment by onox** [comment]
The ideas was that if the option is off, then holding s does what the "Autostart" menu item does (turn on electronics, prime engine, etc.) + enabling starter. If the option is on, then you have to do either turn on electronics + priming yourself, or click "Autostart".

My choice would be to have the option always on and let the "Autostart" menu item be a full autostart instead of the 50% autostart it currently is.

**Comment by onox** [comment]
The Nasal error seems to be solved by moving `<file>Nasal/engine.nas</file>` in `c172p-set.xml` to below `<file>Nasal/c172p.nas</file>`.

**Comment by gilbertohasnofb** [comment]
> The Nasal error seems to be solved by moving <file>Nasal/engine.nas</file> in c172p-set.xml to below <file>Nasal/c172p.nas</file>.

Aha, so that's how it works. That's why `electrical.nas` is able to call a function from `c172p.nas` (the flaps function) but the `engine.nas` wasn't able to do so. I will test this tomorrow and then create a PR if no one beats me to that (which wouldn't be a problem either).

**Comment by onox** [comment]
> My choice would be to have the option always on and let the "Autostart" menu item be a full autostart instead of the 50% autostart it currently is.

@gilbertohasnofb What do you think of this? If you agree you can take a look at the Nasal code in the tutorials for how to programmatically hold the "s" key (enable the starter for a while) and add that to the "Autostart" menu item.

**Comment by dany93** [comment]
@onox wrote
> The ideas was that if the option is off, then holding s does what the "Autostart" menu item does (turn on electronics, prime engine, etc.) + enabling starter. 

Sophisticated ! User-friendly :+1:

**Comment by gilbertohasnofb** [comment]
> The ideas was that if the option is off, then holding s does what the "Autostart" menu item does (turn on electronics, prime engine, etc.) + enabling starter. If the option is on, then you have to either turn on electronics + priming yourself, or click "Autostart". After that you hold s to actually start the engine.

Yes, this is the current behaviour and I think it kind of works.

> My choice would be to have the option always on and let the "Autostart" menu item be a full autostart instead of the 50% autostart it currently is.

So then there would be no simple way of starting the engine except for using autostart? That actually sounds fine to me, given that we currently have two options doing the same thing (autostart and disabling the complex startup) and so I'd be happy to chop one of them off :smile: What I don't understand is what you mean by 50% autostart, which things are our current autostart function missing out?

So basically the PR for this issue would require:

- re-arranging the order of the `.nas` files in teh `-set.xml` file to ensure that the `c172p.nas` comes first and so any of its functions (current or future ones) can be called by other `.nas` files
- remove the option of complex engine startup from GUI, from the `c172p.xml`, etc.
- improve the autostart function

Is that correct?

**Comment by dany93** [comment]
>What I don't understand is what you mean by 50% autostart, which things are our current autostart function missing out?

@onox may mean that with current "Autostart" you also have to press "s" to start.

**Comment by wlbragg** [comment]
>What I don't understand is what you mean by 50% autostart, which things are our current autostart function missing out?

>@onox may mean that with current "Autostart" you also have to press "s" to start.

I have always felt that was kind of odd. Autostart probably should start the aircraft in its entirety.

**Comment by wlbragg** [comment]
So the system might  be
Complex Startup = must do everything
Autostart = starts the aircraft, ready to fly
Complex not checked = requires s key
Start Running = it's running at startup

I really liked the idea of a saved state of "complex" starts you at a hanger or parking spot. Really any choice that is not running should.

**Comment by onox** [comment]
There is no need for a "Complex" checkbutton if we make the "Autostart" menu item a 100 % autostart: people who want to do an autostart, press the "Autostart" menu item. People who do not want to do an autostart, start the aircraft manually.

**Comment by gilbertohasnofb** [comment]
That's all clear then. I will have a look at it and see what I am capable of, ok? I will keep you guys updated.

**Comment by gilbertohasnofb** [comment]
Done. And @onox, great idea to eliminate that option, the whole thing looks so much more clear and simple now. Autostart takes care of everything now (so does starting the sim with the engine running), and all others must operate the aircraft as it was designed to be operated (using prime, etc.)
